2008-02-11  Julian Brown  <julian@codesourcery.com>

        Merge from MIPS:

        2005-03-21  Maciej W. Rozycki  <macro@mips.com>

        * gas/config/tc-mips.c (append_insn): Disable swapping of MIPS16
        branches as it breaks DWARF-2 line information.

--- binutils-2.19.51.0.14/gas/config/tc-mips.c~	2009-06-27 15:47:26.000000000 -0600
+++ binutils-2.19.51.0.14/gas/config/tc-mips.c	2009-07-27 08:20:49.000000000 -0600
@@ -3262,6 +3262,10 @@
 	      /* If the previous instruction is a sync, sync.l, or
 		 sync.p, we can not swap.  */
 	      || (prev_pinfo & INSN_SYNC)
+             /* FIXME: Disable MIPS16 branch swapping for now as it
+                breaks DWARF-2 line information irrecoverably.
+                --macro  */
+              || mips_opts.mips16)
 	      /* If the previous instruction is an ERET or
 		 DERET, avoid the swap.  */
               || (history[0].insn_opcode == INSN_ERET)
